src/bridge/cockpitrouter.c

```diff
@@ -439,19 +481,10 @@ process_open_channel (CockpitRouter *self,
 {
   GType (* type_function) (void) = user_data;
   GType channel_type = 0;
-  const gchar *group;
-
-  if (!cockpit_json_get_string (options, "group", "default", &group))
-    g_warning ("%s: caller specified invalid 'group' field in open message", channel);
 
   g_assert (type_function != NULL);
   channel_type = type_function ();
 
-  if (g_str_equal (group, "fence"))
-    g_hash_table_add (self->fences, g_strdup (channel));
-
-  g_hash_table_insert (self->groups, g_strdup (channel), g_strdup (group));
-
   create_channel (self, channel, options, channel_type);
   return TRUE;
 }
```

Comment:
There's no real meat left in this function any more; but indeed it seems it's mostly just a signature adapter for the `RouterRule->callback`, so I suppose it can't be eliminated completely.